### ISSUE TITLE
Atlas Notifications QA round 2

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/atlas#5181]
# Context

Testing notifications

Previous round of QA
- https://github.com/Joystream/atlas/issues/4891

## Environment to test
- ~~https://atlas-git-fork-thesan-feature-orion-notifications-joystream.vercel.app~~
- ~~https://atlas-git-notifications-joystream.vercel.app~~
- https://dev.gleev.xyz

## Design
[Link to Figma>](https://www.figma.com/file/Goequb9eZDL6WxK7T74oxS/Notifications?type=design&node-id=0-1&mode=design&t=EruFEmBOGyI63AJu-0)

⚠️ Pls check all sub-pages in Figma before starting QA 
<img width="211" alt="Screenshot 2023-10-31 at 12 29 37" src="https://github.com/Joystream/atlas/assets/93924525/5dbc5bac-700c-4dc1-8dbe-f3c867b0eb31">

# Scope
Test that in-app and email notifications are sent to account email address

⚠️ Pls be aware that each individual event will trigger in-app AND email notification right away (no grouping of multiple events to single email is supported)

## Member
- [ ] Channel created
- [ ] Someone replied to your comment
- [ ] Someone reacted to your comment
- [ ] Followed channel posted new video
- [ ] Followed channel put a new nft on auction
- [ ] Followed channel put a new nft on sale
- [ ] NFT bid outbid (someone places a higher bid than your own)
- [ ] NFT open auction bid won
- [ ] NFT open auction bid lost

### Engagement
- [ ] New Channel follower
- [ ] Comment Posted To Video
- [ ] Video Liked
- [ ] Video Disliked

### NFTs
- [ ] NFT purchased
- [ ] NFT bid received (for auction creator)
- [ ] NFT royalty payment received
- [ ] Channel received direct payment (via member remark)
- [ ] Funds withdrawn from channel account (to external account)

### 
- [ ] Notifications for multiple channels 
- [ ] Notifications counters when in member view vs in channel view

### Notification settings
- [ ] Disable some notifications and see if notifications are not been sent as a result of this setting

## Out of scope  
Will be handled by @dmtrjsg 

### YPP
- [x] YPP Channel verified
- [x] YPP Channel suspended

### Content Moderation
- [ ] Channel excluded from App
- [ ] Video is excluded from App
- [x] Video featured as Hero
- [x] Video featured on Category page
- [x] NFT is featured on the Marketplace

**Comment by ivanturlakov** [comment]
@dmtrjsg I got a problem creating a new user, is this related to Orion?
<img width="1439" alt="Screenshot 2023-10-31 at 16 45 20" src="https://github.com/Joystream/atlas/assets/46903215/76209471-33fd-4c72-a8d6-ca6a62d49c7f">

**Comment by dmtrjsg** [comment]
@ivanturlakov yes looks like it, although dev env was reportedly fixed earlier. Let's try and continue tomorrow with that sir 👍

**Comment by dmtrjsg** [comment]
Some comments

1️⃣ Template seems to be a bit off, @KubaMikolajczyk could you pls review and add comments 
ℹ️  [Env to test is here> ](https://atlas-git-fork-thesan-feature-orion-notifications-joystream.vercel.app/)
<img width="529" alt="Screenshot 2023-11-06 at 19 07 22" src="https://github.com/Joystream/atlas/assets/93924525/38405bed-7338-44f9-9b68-d1bffe77d08e">

2️⃣ Unsubscribe should lead to sign in modal, and after signing in, transfer to notifications settings (on channel or member, depending on the email sent)

**Comment by ivanturlakov** [comment]
Recipient name is not correct

Current recipient=**notifqa2**
<img width="619" alt="Screenshot 2023-11-07 at 12 42 29" src="https://github.com/Joystream/atlas/assets/46903215/106f3608-a439-4539-a066-8bc0e4634b14">

**Comment by ivanturlakov** [comment]
Tested on https://atlas-git-fork-thesan-feature-orion-notifications-joystream.vercel.app
https://atlas-git-notifications-joystream.vercel.app/

### Member

✅ Channel created
⚠️ Someone replied to your comment(Atlas - notification goes to author of Reply instead of author of Comment, Email - Ok)
✅ Someone reacted to your comment
✅ Followed channel posted new video(⚠️ Video didn’t uploaded(failed) but notification sent)
✅ Followed channel put a new nft on auction
✅ Followed channel put a new nft on sale
⚠️ NFT bid outbid (someone places a higher bid than your own) - no notifications at all(App, Email)
⚠️ NFT open auction bid won/lost - Email - OK, App - error

https://github.com/Joystream/atlas/assets/46903215/23be75f4-6e2c-4a99-96ef-670db545157d



### NFT
 ✅ NFT purchased
 ✅ NFT bid received (for auction creator)
 ✅ NFT royalty payment received
 🚫 Channel received direct payment (via member remark) - blocked, pending instructions how to pay channels on dev
 ✅ Funds withdrawn from channel account (to external account)
⚠️ UI shows - Something goes wrong, but the transaction is successful and channel balance is updated
<img width="1439" alt="Screenshot 2023-11-10 at 16 49 37" src="https://github.com/Joystream/atlas/assets/46903215/f60b86a8-742d-4e2e-83e6-05ebe754f32b">
⚠️ destination address not specified, pic missed
<img width="582" alt="Screenshot 2023-11-10 at 16 50 52" src="https://github.com/Joystream/atlas/assets/46903215/24cede7f-a9b4-49d4-b98b-eda6b747f49e">
<img width="1038" alt="Screenshot 2023-11-09 at 16 30 04" src="https://github.com/Joystream/atlas/assets/46903215/36522234-ed87-499b-8989-2f3cc90eb52d">

⚠️ Notifications for multiple channels
There is a problem with the counter of unread notifications; when opened, it is not updated correctly. At the very beginning, I had 3 notifications in the studio (1st channel - 2, 2nd channel - 1)

https://github.com/Joystream/atlas/assets/46903215/4ab79377-15c9-495d-a0d4-0645b40e2796


 ✅ Notifications counters when in member view vs in channel view - looks intuitive
**Atlas view**
Someone reacted to your comment
<img width="150" alt="Screenshot 2023-11-10 at 17 02 55" src="https://github.com/Joystream/atlas/assets/46903215/a269032c-3420-4881-ae58-19db8d68e98c">
<img width="316" alt="Screenshot 2023-11-10 at 17 03 14" src="https://github.com/Joystream/atlas/assets/46903215/4603e448-55a9-4e98-a0c1-9601cad915f0">
+Comment Posted To Video
<img width="143" alt="Screenshot 2023-11-10 at 17 03 33" src="https://github.com/Joystream/atlas/assets/46903215/b31d0b05-d20c-4be5-97d2-17023911149c">
<img width="326" alt="Screenshot 2023-11-10 at 17 03 47" src="https://github.com/Joystream/atlas/assets/46903215/9e66023a-1f11-4cac-9655-a964704ed972">
Switch your to  **Studio view**
<img width="297" alt="Screenshot 2023-11-10 at 17 04 06" src="https://github.com/Joystream/atlas/assets/46903215/0c804a3d-ec5a-47a8-ac04-13f389b6c3b0">
<img width="312" alt="Screenshot 2023-11-10 at 17 04 18" src="https://github.com/Joystream/atlas/assets/46903215/537f72d3-93d2-4a91-8f2e-375bdc3174ea">

⚠️ Pics missed & number format **.00**
<img width="337" alt="Screenshot 2023-11-10 at 17 06 42" src="https://github.com/Joystream/atlas/assets/46903215/ef9d7674-1022-438e-9f75-e7546ca55157">
<img width="330" alt="Screenshot 2023-11-10 at 17 06 21" src="https://github.com/Joystream/atlas/assets/46903215/5417b877-86e6-4c6d-8b72-52d9e70e67cc">
<img width="395" alt="Screenshot 2023-11-10 at 16 16 13" src="https://github.com/Joystream/atlas/assets/46903215/794199b3-965f-4f5d-9fa2-42de61085e80">
<img width="579" alt="Screenshot 2023-11-10 at 16 16 45" src="https://github.com/Joystream/atlas/assets/46903215/3e9e07fe-0ee8-4260-9ad3-36ada52794fd">
<img width="398" alt="Screenshot 2023-11-10 at 16 15 51" src="https://github.com/Joystream/atlas/assets/46903215/7d656690-7a18-4f2a-af27-2140469f3ade">
<img width="577" alt="Screenshot 2023-11-10 at 16 11 41" src="https://github.com/Joystream/atlas/assets/46903215/2a3d302d-9db3-445d-a218-fba0c10eb769">


### Channel
**Engagement**
✅ New Channel follower
✅ Comment Posted To Video
✅ Video Liked
✅ Video Disliked


⚠️ The notification counter at each login shows that I have 7 unread notifications (actually there are 7 in total, 3 are unread). !This only happens when switching between users.


**Notification settings**
(Disable some notifications and see if notifications are not been sent as a result of this setting)
✅ Someone reacted to your comment
✅ Someone replied to your comment
⚠️ Followed channel posted new video(App notification didn't come when Email notification was disabled)
⚠️ Followed channel put a new nft on sale(App notification didn't come when Email notification was disabled)
⚠️ Followed channel put a new nft on auction(Email notification didn't come when App notification was disabled)

**Comment by ivanturlakov** [comment]
⬆️ UPD

**Comment by dmtrjsg** [comment]
@ivanturlakov thank you sir for the comprehensive testing. We will fix and tag you for the round 3 🙏

**Comment by dmtrjsg** [comment]
Tested @thesan  @ignazio-bovo 

1. YPP verification of channels  / also for suspended - ⛔ no in-app, and no email
2. Excluded video ⛔ - no in-apps, and no email
3. Member payment - ✅ in app okl, ⚠️ email notification received, but CTA pointed to page that did not resolve properly url `https://gleev.xyz/studio/notifications/channel`
<img width="952" alt="Screenshot 2023-11-15 at 19 06 57" src="https://github.com/Joystream/atlas/assets/93924525/0ddad652-1fa6-4d18-a021-5c26d83c3ca3">

**Comment by thesan** [comment]
>   3. Member payment - ✅ in app okl, ⚠️ email notification received, but CTA pointed to page that did not resolve properly url `https://gleev.xyz/studio/notifications/channel`

`https://gleev.xyz/studio/notifications/channel` won't resolve until the notifications are release to production. But `https://dev.gleev.xyz/studio/notifications/channel` does if you're singed in

**Comment by thesan** [comment]
@ivanturlakov thank you detailed testing.

> Notifications for multiple channels
> There is a problem with the counter of unread notifications; when opened, it is not updated correctly. At the very beginning, I had 3 notifications in the studio (1st channel - 2, 2nd channel - 1)

 Maybe I'm missing something but I haven't been able to reproduce this so far. Could you try to reproduce it again and give the steps to do so please ?

**Comment by thesan** [comment]
@ivanturlakov never mind I just did !

**Comment by dmtrjsg** [comment]
@thesan @ignazio-bovo  
Guys I finished testing the content moderation: 

- [x] Featured NFTs ✅  in email and in-app
- [ ] Channel excluded ⛔

**Comment by dobertRowneySr** [comment]
Channel excluded works for in app and email. 
Maybe the only detail would be change text to "got excluded from Gleev" instead of "got excluded from App".
And the email channel title

<img width="866" alt="image" src="https://github.com/Joystream/atlas/assets/51839739/6ee9549b-91b7-4371-98df-8b43f041964d">
<img width="678" alt="image" src="https://github.com/Joystream/atlas/assets/51839739/fb828b03-283e-4e9d-a315-51f5ec557dc5">

**Comment by dmtrjsg** [comment]
@ivanturlakov sir, everything from [this comment](https://github.com/Joystream/atlas/issues/5181#issuecomment-1798456261) is ready to be re-tested.

**Comment by thesan** [comment]
@ivanturlakov I just updated the environment part of the issue. Sorry again for not mentioning it earlier.

**Comment by ivanturlakov** [comment]
In progress... 

Tested on https://dev.gleev.xyz/

⚠️ Member notifications counter issue

The counter does not reset after viewing notifications

https://github.com/Joystream/atlas/assets/46903215/f8a35b13-1c25-4ba4-8fda-36c6db361b8f

https://github.com/Joystream/atlas/assets/46903215/4849b513-a2db-4f05-975c-c40aa8a484a6

✅ Someone replied to your comment
⚠️ NFT bid outbid (someone places a higher bid than your own) - no notifications at all(App, Email)
✅ NFT open auction bid won/lost
✅ Funds withdrawn from channel account (to external account)
<img width="582" alt="Screenshot 2023-11-21 at 15 47 16" src="https://github.com/Joystream/atlas/assets/46903215/342a0365-a5e1-4117-b7e9-fd42fc3decd5">
<img width="1314" alt="Screenshot 2023-11-21 at 15 48 08" src="https://github.com/Joystream/atlas/assets/46903215/909c828d-b190-48ac-bc9c-31da9a2bca4c">

✅ Notifications for multiple channels

⚠️ No chance to upload the video to proceed
<img width="1440" alt="Screenshot 2023-11-21 at 16 26 17" src="https://github.com/Joystream/atlas/assets/46903215/6764d015-1e3b-43e0-9b82-e5a04dfe2e73">

**Notification settings**
(Disable some notifications and see if notifications are not been sent as a result of this setting)
Followed channel posted new video(App notification didn't come when Email notification was disabled)
Followed channel put a new nft on sale(App notification didn't come when Email notification was disabled)
Followed channel put a new nft on auction(Email notification didn't come when App notification was disabled)

**Comment by attemka** [comment]
@ivanturlakov upload bug was fixed on dev env, so this part can be retested. 
@ignazio-bovo could you assist on the NFT outbid please?

**Comment by dobertRowneySr** [comment]
As suspected:
`Higher Bid Placed` notification works only for timed auctions.
hence the only change I would add is on the notification text:
```
        text: `${newBidderHandle} placed a higher bid in the timed auction for NFT: “${videoTitle}”`,
```

**Comment by dmtrjsg** [comment]
@ivanturlakov pls feel free to retest 
- notification counters
- video reactions and all cases connected with uploads

**Comment by ivanturlakov** [comment]
Tested on https://dev.gleev.xyz/

✅ Notification Сounter behavior
✅ Uploads/NFT Mint/NFT Sale
✅ NFT bid outbid for timed auction (someone places a higher bid than your own)
<img width="578" alt="Screenshot 2023-11-23 at 14 25 40" src="https://github.com/Joystream/atlas/assets/46903215/d9081b3f-7a3d-449c-bb2f-d3f19f567b8c">
<img width="380" alt="Screenshot 2023-11-23 at 14 26 28" src="https://github.com/Joystream/atlas/assets/46903215/32517eba-6f55-41aa-8e35-2790fcd22d60">

**Notification settings**
(Disable some notifications and see if notifications are not been sent as a result of this setting)
✅ Someone replied to your comment
✅ Someone reacted to your comment
✅ Followed channel posted new video
✅ Followed channel put a new nft on sale
✅ Followed channel put a new nft on auction

Some pics still didn't show up
<img width="576" alt="Screenshot 2023-11-23 at 13 30 49" src="https://github.com/Joystream/atlas/assets/46903215/c93e5aed-c45e-47d1-8e87-8149023f59a2">
<img width="577" alt="Screenshot 2023-11-23 at 13 31 27" src="https://github.com/Joystream/atlas/assets/46903215/017b9dae-738c-400e-903c-c1d2f7bda820">
<img width="574" alt="Screenshot 2023-11-23 at 13 31 35" src="https://github.com/Joystream/atlas/assets/46903215/93d57fe5-6aa4-4fcf-900a-cb7a5f513cb2">
<img width="578" alt="Screenshot 2023-11-23 at 14 03 32" src="https://github.com/Joystream/atlas/assets/46903215/7b3ecb73-6a01-4c72-9c06-c9518df25be1">

**Comment by ivanturlakov** [comment]
I accidentally liked my own comment and received a notification in the app and by email 🙂 This somehow confused the counter and for some time it showed the incorrect amount (1 more), after re-login everything was fine

**Comment by dmtrjsg** [comment]
Thank you @ivanturlakov for the excellent QA work. On the basis of your report we are taking the decision to release to prod 👍 

For the last two reported issues we will address them as a follow up.
